### ISSUE TITLE
Handle date types correctly in server response, as well as server request if applicable.

### ISSUE THREAD

**Issue by gregoryterlecki** [gregoryterlecki/typescript-app-demo#2]
## Problem
TypeScript error when typing client data using types from server side router.

## Details
_Currently_, in the case that you pass data containing dates from a server parent component to a client child component, you will get a typescript error.

The server is not able to send `Date` types as a response back to your client; at least not in the current state of the codebase. This means we'll get typescript errors, since the trpc react client.

Keep in mind this problem might very well be highly specific to the problem it was encountered in.
Basically, don't use the router type definitions from `serverClient` when specifying `initialData` to `useQuery();`. You will get TS errors. See if you can get more accurate typing from `_trpc/client`.
